server/src/server.js

```diff
@@ -49,6 +49,10 @@ app.use(function (req, res, next) {
 app.use(function (req, res, next) {
   req.staticLink = linker.staticLink;
   req.staticLinkWithHost = linker.staticLinkWithHost.bind(null, req);
+  if (Shot.prototype.imageLink === undefined) {
```

Comment:
Yes, it's not supposed go on the prototype, whoops. (How is this working? I guess because the url just starts with / and not http?)

Yes, it does mean the first request defines the host for the life of the process. Do you have a better idea? The only other idea I had was to put it in a configuration file, which isn't the greatest.